framework/wazuh/agent.py

```diff
@@ -443,89 +443,73 @@ def _remove_manual(self, backup=False, purge=False):
         :param purge: Delete definitely from key store.
         :return: Message.
         """
-
-        # Get info from DB
+        # Check if agent exists
         self._load_info_from_DB()
 
         f_keys_temp = '{0}.tmp'.format(common.client_keys)
-        open(f_keys_temp, 'a').close()
-
-        f_keys_st = stat(common.client_keys)
-        chown(f_keys_temp, common.ossec_uid, common.ossec_gid)
-        chmod(f_keys_temp, f_keys_st.st_mode)
-
-        f_tmp = open(f_keys_temp, 'w')
-        agent_found = False
-        with open(common.client_keys) as f_k:
-            for line in f_k.readlines():
-                line_data = line.strip().split(' ')  # 0 -> id, 1 -> name, 2 -> ip, 3 -> key
-
-                if self.id == line_data[0] and line_data[1][0] not in ('#!'):
-                    if not purge:
-                        f_tmp.write('{0} !{1} {2} {3}\n'.format(line_data[0], line_data[1], line_data[2], line_data[3]))
-
-                    agent_found = True
-                else:
-                    f_tmp.write(line)
-        f_tmp.close()
-
-        if not agent_found:
-            remove(f_keys_temp)
-            raise WazuhException(1701, self.id)
-
-        # Overwrite client.keys
-        move(f_keys_temp, common.client_keys)
-
-        # Remove rid file
-        rids_file = '{0}/queue/rids/{1}'.format(common.ossec_path, self.id)
-        if path.exists(rids_file):
-            remove(rids_file)
 
-        # get agent's group
-        group_name = ""
-        group_path = "{}/{}".format(common.groups_path, self.id)
-
-        if path.exists(group_path):
-            with open(group_path) as f:
-                group_name = f.read().replace('\n', '')
-                f.close()
+        try:
+            agent_found = False
+            with open(common.client_keys) as client_keys, open(f_keys_temp, 'w') as client_keys_tmp:
```

Comment:
What happens if some kind of Exception is raised? I think `f_keys_temp` would remain in the disk and never be deleted. I suggest adding a `finally` block to ensure all temp files are removed smartly.

Reply:
Fixed in https://github.com/wazuh/wazuh/pull/2966/commits/4d8e304957ebc722a19e2a015d4f114f34c0f150